include/s3select_oper.h

```diff
@@ -569,6 +625,10 @@ class value
 
   bool operator<(const value& v)//basic compare operator , most itensive runtime operation
   {
+    if ((is_null() || v.is_null()) || (is_nan() || v.is_nan()))
```

Comment:
as for optimization,
the nan/null cases are rare compare to others, 
thus, its possible to "push them" down, and save some instructions per comparison.